### PR TITLE
Filter products

### LINKED ISSUE #14: Filter products
## Value statement - a requirement, not a solution
As a **user** 
I want **a feature to filter the products** 
so that **I can hide products that I am allergic to** 

## Description (Text, Scribble, Wireframe, Design)
Screenshot 2020-09-16 at 11 44 24

## Acceptance criteria
- [ ] Filter show on each product list page
- [ ] Filter contains the allergies "gluten" and "soy"
- [ ] By clicking on one of the allergies all products with these allergies are hidden

## Tasks
- [ ] Create feature branch
- [ ] Add filter icon to product list page
- [ ] Create filter component in storybook
- [ ] Use filter method on products
- [ ] Test the component with unit and integration tests
- [ ] Implement into App

## Size
medium

### PR DESCRIPTION
**Pull request by denniswitzel**
This pull request links to the issue #14 

Screenshot 2020-09-16 at 11 44 24

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,88 +3,52 @@ import { Route, Switch } from 'react-router-dom'
...
       <Route exact path="/">
         <LogoStyled />
-        {categories.map((category) => (
-          <Categories
-            key={category.name}
-            productIcon={category.icon}
-            productName={category.name}
-          />
+        {categories.map(({name, icon}) => (
+          <Categories key={name} categoryIcon={icon} categoryName={name}/>
```
Comment (LebkuchenCJ) [inline src/App.js]: Couldn`t name be tricky when you add more products? Do have access to any ID?

File: src/components/Filter/Filter.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    grid-column: 1/3;
+    display: flex;
+    margin-bottom: 20px;
+`
+
+const FilterButton = styled.button`
+    width: 85px;
+    height: 35px;
+    border: 2px solid #4BDB80;
+    color: ${({active}) => active ? 'white' : '#4BDB80'};
```
Comment (LebkuchenCJ) [inline src/components/Filter/Filter.js]: You could stay with one convention
```suggestion
    color: ${({active}) => active ? '#FFF' : '#4BDB80'};
```

File: src/components/Filter/Filter.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+const ButtonWrapper = styled.section`
+    grid-column: 1/3;
+    display: flex;
+    margin-bottom: 20px;
+`
+
+const FilterButton = styled.button`
+    width: 85px;
```
Comment (LebkuchenCJ) [inline src/components/Filter/Filter.js]: Maybe % or rem works better for a more responsive design

File: src/components/ProductDetail/ProductDetail.js [hunk 4]
```diff
@@ -1,55 +1,58 @@
...
+        {productItem.allergies?.length ? (
           <Headline>Allergies</Headline>
         ) : (
           <Headline></Headline>
         )}
-        {product.allergies?.map((allergy, index) => (
+        {productItem.allergies?.map((allergy, index) => (
           <AllergyContainer key={index}>
-            {allergy === 'Gluten' ? <Gluten /> : <Soy />}
+            {allergy === 'gluten' ? <Gluten /> : <Soy />}
```
Comment (LebkuchenCJ) [inline src/components/ProductDetail/ProductDetail.js:48]: I'm not into allergies but wouldn't it be a nice option to hide both? Or does that not make any sense? :)

File: src/components/ProductDetail/ProductDetail.js [hunk 5]
```diff
@@ -126,11 +129,12 @@ const Allergies = styled.p`
   font-weight: 300;
   text-align: left;
   margin: 0;
+  text-transform: capitalize;
 `
 const ArrowStyled = styled(Arrow)`
   fill: #ffffff;
   position: absolute;
-  left: 30px;
+  left: 50px;
```
Comment (LebkuchenCJ) [inline src/components/ProductDetail/ProductDetail.js]: % or rem?

File: src/components/Products/Product.js [hunk 6]
```diff
@@ -55,4 +41,8 @@ const ProductImage = styled.img`
   margin-top: 10px;
   -webkit-filter: drop-shadow(2px 2px 6px rgba(0, 0, 0, 0.2));
   filter: drop-shadow(2px 2px 6px rgba(0, 0, 0, 0.2));
+
+  :last-child {
+    margin-bottom: 110px;
```
Comment (LebkuchenCJ) [inline src/components/Products/Product.js]: % or rem?

**Review by LebkuchenCJ — APPROVED** [review]
Hey, nice filter. GJ 👍 
Only found some small style related issues. Happy hacking 💯

File: src/App.js (hunk 1 above)
Comment (denniswitzel) [reply src/App.js]: Good call! I'm gonna change that

**Pull request merged**